configure.ac

```diff
@@ -2630,11 +2630,11 @@ esac
 # add wolfSSL defines for shared/static
 if test "$enable_shared" = "yes"
 then
-    AM_CFLAGS="$AM_CFLAGS -DWOLFSSL_SHARED"
+    AC_DEFINE([WOLFSSL_SHARED], [1], [Define if shared library avialble.])
 fi
 if test "$enable_static" = "yes"
 then
-    AM_CFLAGS="$AM_CFLAGS -DWOLFSSL_STATIC"
+    AC_DEFINE([WOLFSSL_STATIC], [1], [Define if static library avialble.])
```

Comment:
This moves WOLFSSL_STATIC and WOLFSSL_SHARED macros to config.h which is the same location that HAVE_VISIBILITY macro is. Still allowing for a record of which type is used but keeping the information local to the build.